packages/runtime/container-runtime/src/garbageCollectionTombstoneUtils.ts

```diff
@@ -3,29 +3,21 @@
  * Licensed under the MIT License.
  */
 
-import { ITelemetryGenericEvent } from "@fluidframework/common-definitions";
+import { ITelemetryGenericEvent, ITelemetryLogger } from "@fluidframework/common-definitions";
 import { packagePathToTelemetryProperty } from "@fluidframework/runtime-utils";
-import { MonitoringContext } from "@fluidframework/telemetry-utils";
-import { throwOnTombstoneUsageKey } from "./garbageCollectionConstants";
 
-/**
- * Decides whether or not to send an error event or a generic event for gc tombstone scenarios
- *
- * Adds isSummarizerClient, packagePath, and error to telemetry properties.
- */
+/** Augments and sends the given event/error via the given logger */
 export function sendGCTombstoneEvent(
-    mc: MonitoringContext,
+    logger: ITelemetryLogger,
     event: ITelemetryGenericEvent,
+    logAsError: boolean,
     isSummarizerClient: boolean,
     packagePath: readonly string[] | undefined,
-    error?: any,
+    error: unknown,
 ) {
-    const throwOnTombstoneUsage = event.throwOnTombstoneUsage = mc.config.getBoolean(throwOnTombstoneUsageKey) ?? false;
+    event.category = logAsError ? "error" : "generic";
```

Comment:
nit: It bugs me that this function accepts event along with properties that are added to event. It would be better to pass in all the properties directly and create an event in this function.